app/policies/comment_policy.rb

```diff
@@ -1,28 +1,20 @@
 class CommentPolicy < ApplicationPolicy
-  def new?
-    company = @record.company
-    user.present? &&
-      (company.owner == @user.user || company.authors.include?(@user.user))
-  end
-
   def create?
-    company = @record.company
-    company.owner == @user || company.authors.include?(@user)
-  end
-
-  def edit?
-    manage?
+    user.present? &&
+      (company.owner == user || company.authors.include?(user))
```

Comment:
https://semaphoreci.com/blog/2017/03/14/faster-rails-how-to-check-if-a-record-exists.html
`.exists?` is faster than `.include?`
```suggestion
      (company.owner == user || company.authors.exists?(user.id))
```

Reply:
seems like, `company.owner == user` is use in few places. I think we can move this check to a separate method: 
```ruby
def company_owner?
  @company_owner ||= company.owner == user
end
```